### DISCUSSION TITLE
Clearance parameter in RSA (and other sphere inserting methods) not available in version < 2.0?

### DISCUSSION THREAD

**Discussion by Olga5991** [PMEAL/porespy#544] [Q&A] [answered]
Hello all,

I was trying to use RSA generator with keyword 'clearance', but it seems that RSA doesn't have this argument. Maybe I should use another version of porespy? Installed version is 1.3.1. Is there any other way to generate random distribution of non-overlapping and non-tangent spheres with different radii? Thanks in advance!

**Comment by jgostick** [comment, marked as answer]
Yes, you need to be using version 2.  The docs website is generated from V2, but we have been dragging our feet on releasing the official version because it will then cause conflicts with openpnm (via the network extraction algorithms, which output slightly different values in PoreSpy V2).  So we are not sure which is worse, delaying the release of PoreSpy V2, or breaking the compatibility with OpenPNM.  We are working on OpenPNM V3, but it's looking that that will take quite some time.  

To get PoreSpy V2 there are a few options:

1: The basic way: Download the source code from Github, unzip it and move it somewhere of your choice, then open your conda prompt in the new location (in the same folder as the setup.py file), then run ``pip install -e .``  This will install the porespy into your conda environment.  I just tested this and it automatically uninstalled my old version and added the new one (which does nothing in my case since they are the same version)
2: The correct way: Open a git client (fork, gitkraken, sourcetree, etc), and clone the repo using "https://github.com/PMEAL/porespy.git".  This has the same effect as downloading the code, except that your git client will be aware of the repo and show you any changes.  Then "checkout" the dev branch by double clicking it in the gui.  Then proceed as above starting the "open your conda prompt" part. Using this approach means that you can continually "pull" new changes down to your local machine and they will automatically be loaded next time you use porespy.  You can also "checkout" other versions, such 1.3.1 for instance, just by double clicking on the branch or tag.

**Reply by Olga5991** [reply]
Thank you very much!

**Reply by jgostick** [reply]
I hope it works...if not, let me know and I'll amend my answer
